src/machine/machine_avr.go

```diff
@@ -58,3 +58,93 @@ func (p GPIO) Get() bool {
 		return (val > 0)
 	}
 }
+
+const (
+	COM2A1 = 7
+	COM2A0 = 6
+	COM2B1 = 5
+	COM2B0 = 4
+	COM0A1 = 7
+	COM0A0 = 6
+	COM0B1 = 5
+	COM0B0 = 4
+	COM1A1 = 7
+	COM1A0 = 6
+	COM1B1 = 5
+	COM1B0 = 4
+
+	CS00 = 0
+	CS01 = 1
+	CS11 = 1
+	CS22 = 2
+
+	WGM10 = 0
+	WGM11 = 1
+	WGM20 = 0
+	WGM21 = 1
```

Comment:
These values should ideally not be hardcoded here, but automatically generated from the .atdf file of this device. The source is located at `lib/avr/packs/atmega/ATmega328P.atdf` and the script to generate the bitfields is `tools/gen-device-avr.py`.
It will probably require some guesswork, trial and error to get them right, but it will avoid having to hardcode constants in the source here.

Also, I generally avoid specifying literal bit numbers as is custom for the AVRs because it leads to a lot of code like `(1 << foo)`, see for example the code in `src/runtime/runtime_avr.go` that avoids it. I would instead prefer the actual values so you can use them directly (e.g. simply `*avr.TCCR0B |= avr.CS01`). Unless, of course, you know a good reason not to do it like this.

Reply:
I dug thru the atdf files for a while before I added those constants to `machine_avr.go`. None of them appear to be values that match the section that seem like they should belong to. For example `TCCR0B` we need to set to 0x03. Using your suggested syntax:

```go
*avr.TCCR0B |= (1 << CS01) | (1 << CS00)
```

However the atdf file does not have a value that corresponds to 0x03 for the `TCCR0B` register:

```xml
<register caption="Timer/Counter Control Register B" name="TCCR0B" offset="0x45" size="1">
    <bitfield caption="Force Output Compare A" mask="0x80" name="FOC0A"/>
    <bitfield caption="Force Output Compare B" mask="0x40" name="FOC0B"/>
    <bitfield caption="" mask="0x08" name="WGM02"/>
    <bitfield caption="Clock Select" mask="0x07" name="CS0" values="CLK_SEL_3BIT_EXT"/>
</register>
```

I found these values by digging thru the Arduino AVR source code https://github.com/arduino/ArduinoCore-avr/blob/master/cores/arduino/wiring_analog.c and sone AVR docs such as https://sites.google.com/site/qeewiki/books/avr-guide/pwm-on-the-atmega328

Anyhow, there does not seem to be a way to parse the values if they are not present in the atdf file. What do you suggest?

Reply:
I *think* the extra number at the end is the 0-indexed bit number. So for example, there is `COM2A` with value 0xC0 in the .atdf file, which is 11000000 in binary. So there are two bits in this part of the register. This matches with the two bits of the register that are specified in the C header files: `COM2A0` and `COM2A1`.
The same goes for `CS0`, which has 3 bits (0x07) and has values `CS00`, `CS01` and `CS02` (with values 0, 1, and 2).

Reply:
Updated `gen-device-avr.py` based on your feedback.